src/traces/sankey/render.js

```diff
@@ -155,7 +155,7 @@ function linkModel(uniqueKeys, d, l) {
     var tc = tinycolor(l.color);
     var basicKey = l.source.label + '|' + l.target.label;
     var foundKey = uniqueKeys[basicKey];
-    uniqueKeys[basicKey] = (foundKey === void(0) ? foundKey : 0) + 1;
+    uniqueKeys[basicKey] = (foundKey === void(0) ? 0 : foundKey) + 1;
```

Comment:
@alexcjohnson yes, the 1st suggestion works out if we 1-index things as you do, and indeed there is a small risk of name collision - commit on the way